### PR TITLE
Exclude `org.testcontainers.shaded.*` package and upgrade deps

### PR DESCRIPTION
**Pull request by eddumelendez**
Add general japicmp's package exclusion.

Also, shadow and test dependencies below are upgraded:
* Upgrade okhttp version to 4.10.0
* Upgrade javassist version to 3.29.0-GA
* Upgrade zip4j version to 2.11.1
* Upgrade kubernetes-client version to 6.0.0

### CODE REVIEW

File: modules/azure/build.gradle [hunk 1]
```diff
@@ -1,9 +1,15 @@
 description = "Testcontainers :: Azure"
 
+tasks.japicmp {
+    packageExcludes = [
+        "org.testcontainers.shaded.*",
```
Comment (bsideup) [inline modules/azure/build.gradle:5]: We should make this exclusion global in gradle/japicmp.gradle 👍

Comment (eddumelendez) [reply modules/azure/build.gradle]: done

File: modules/mockserver/src/test/java/org/testcontainers/containers/MockServerContainerRuleTest.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
 public class MockServerContainerRuleTest {
 
     public static final DockerImageName MOCKSERVER_IMAGE = DockerImageName.parse(
-        "jamesdbloom/mockserver:mockserver-5.5.4"
+        "jamesdbloom/mockserver:mockserver-5.13.2"
```
Comment (bsideup) [inline modules/mockserver/src/test/java/org/testcontainers/containers/MockServerContainerRuleTest.java:15]: Just curious why we had to update this one?

**Review by bsideup — APPROVED** [review]
👍

File: modules/mockserver/src/test/java/org/testcontainers/containers/MockServerContainerRuleTest.java (hunk 2 above)
Comment (eddumelendez) [reply modules/mockserver/src/test/java/org/testcontainers/containers/MockServerContainerRuleTest.java:15]: mockserver dependency expects two additional fields that are not provided by an old image. I remember that it is recommended to use the same version for the jar and docker image

**Pull request merged**